src/ape/api/transactions.py

```diff
@@ -151,14 +151,21 @@ class ReceiptAPI(BaseInterfaceModel):
     logs: List[dict] = []
     status: int
     txn_hash: str
-    transaction: TransactionAPI
+    transaction: TransactionAPI | dict
 
     def __repr__(self) -> str:
         return f"<{self.__class__.__name__} {self.txn_hash}>"
 
     def __getattr__(self, item: str) -> Optional[Any]:
         return getattr(self.transaction, item)
 
+    def __post_init__(self):
+        if isinstance(self.transaction, dict):
+            self.transaction = self.create_transaction(self.transaction)
```

Comment:
You should be able to do this as a validator instead

Reply:
Will do tonight

Reply:
@NotPeopling2day's example, but first check if it's a `dict` then do `TransactionAPI.parse_obj`, else just return `v`